cranelift/codegen/src/machinst/lower.rs

```diff
@@ -386,7 +387,7 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
 
         // Assign vreg(s) to each return value.
         let mut retval_regs = vec![];
-        for ret in &f.signature.returns {
+        for ret in &vcode.abi().signature().returns.clone() {
```

Comment:
Is the clone necessary here?

Reply:
Unfortunately yes, otherwise this holds an immutable borrow on `vcode` for the duration of the loop while we also pass in `&mut vcode` to `alloc_vregs()` below. I considered turning some subset of the vcode inside-out somehow to work around this but it didn't seem worth it...